cmd/nerdctl/update.go

```diff
@@ -0,0 +1,256 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"runtime"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/containers"
+	"github.com/containerd/containerd/errdefs"
+	"github.com/containerd/nerdctl/pkg/formatter"
+	"github.com/containerd/nerdctl/pkg/idutil/containerwalker"
+	"github.com/containerd/typeurl"
+	"github.com/docker/go-units"
+	runtimespec "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+type updateOptions struct {
+	CpuPeriod uint64
+	// CPU CFS (Completely Fair Scheduler) quota. Default: 0 (not specified).
+	CpuQuota int64
+	// CPU shares (relative weight vs. other containers). Default: 0 (not specified).
+	CpuShares uint64
+	// Memory limit in bytes. Default: 0 (not specified).
+	MemoryLimitInBytes int64
+	// CpusetCpus constrains the allowed set of logical CPUs. Default: "" (not specified).
+	CpusetCpus string
+	// CpusetMems constrains the allowed set of memory nodes. Default: "" (not specified).
+	CpusetMems string
+}
+
+func newUpdateCommand() *cobra.Command {
+	var updateCommand = &cobra.Command{
+		Use:               "update [flags] CONTAINER [CONTAINER, ...]",
+		Args:              cobra.MinimumNArgs(1),
+		Short:             "Update one or more running containers",
+		RunE:              updateAction,
+		ValidArgsFunction: updateShellComplete,
+		SilenceUsage:      true,
+		SilenceErrors:     true,
+	}
+	updateCommand.Flags().SetInterspersed(false)
+	setUpdateFlags(updateCommand)
+	return updateCommand
+}
+
+func setUpdateFlags(cmd *cobra.Command) {
+	cmd.Flags().Float64("cpus", 0.0, "Number of CPUs")
+	cmd.Flags().Uint64("cpu-period", 0, "Limit CPU CFS (Completely Fair Scheduler) period")
+	cmd.Flags().Int64("cpu-quota", -1, "Limit CPU CFS (Completely Fair Scheduler) quota")
+	cmd.Flags().Uint64("cpu-shares", 0, "CPU shares (relative weight)")
+	cmd.Flags().StringP("memory", "m", "", "Memory limit")
+	cmd.Flags().String("cpuset-cpus", "", "CPUs in which to allow execution (0-3, 0,1)")
+	cmd.Flags().String("cpuset-mems", "", "MEMs in which to allow execution (0-3, 0,1)")
+}
```

Comment:
Please update the Command Reference documentation like this

https://github.com/containerd/nerdctl/commit/19ca8c36a1f7e2e3bbc0e48a0d58954573f111f6